packages/in-app-notification-dispatcher/src/index.ts

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable sonarjs/no-identical-functions */
+import { runConsumer } from "kafka-iam-auth";
+import { EachMessagePayload } from "kafkajs";
+import { decodeKafkaMessage, logger, Logger } from "pagopa-interop-commons";
+import {
+  AgreementEventEnvelopeV2,
+  AgreementEventV2,
+  CorrelationId,
+  EServiceEventEnvelopeV2,
+  EServiceEventV2,
+  generateId,
+  genericInternalError,
+  missingKafkaMessageDataError,
+  PurposeEventEnvelopeV2,
+  PurposeEventV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
+import {
+  agreementReadModelServiceBuilder,
+  catalogReadModelServiceBuilder,
+  makeDrizzleConnection,
+  tenantReadModelServiceBuilder,
+} from "pagopa-interop-readmodel";
+import { drizzle } from "drizzle-orm/node-postgres";
+import pg from "pg";
+
+// Config & Services
+import { config } from "./config/config.js";
+import { readModelServiceBuilderSQL } from "./services/readModelServiceSQL.js";
+import { inAppNotificationServiceBuilderSQL } from "./services/inAppNotificationServiceSQL.js";
+
+// Handlers
+import * as handlers from "./handlers/index.js";
+
+interface TopicHandlers {
+  catalogTopic: string;
+  agreementTopic: string;
+  purposeTopic: string;
+}
+
+const readModelDB = makeDrizzleConnection(config);
+const agreementReadModelServiceSQL =
+  agreementReadModelServiceBuilder(readModelDB);
+const catalogReadModelServiceSQL = catalogReadModelServiceBuilder(readModelDB);
+const tenantReadModelServiceSQL = tenantReadModelServiceBuilder(readModelDB);
+
+const readModelService = readModelServiceBuilderSQL({
+  agreementReadModelServiceSQL,
+  catalogReadModelServiceSQL,
+  tenantReadModelServiceSQL,
+});
+
+const notificationDB = drizzle(
+  new pg.Pool({
+    host: config.inAppNotificationDBHost,
+    database: config.inAppNotificationDBName,
+    user: config.inAppNotificationDBUsername,
+    password: config.inAppNotificationDBPassword,
+    port: config.inAppNotificationDBPort,
+  })
+);
+
+const inAppNotificationService =
+  inAppNotificationServiceBuilderSQL(notificationDB);
+
+export async function handleCatalogMessage(
+  decodedMessage: EServiceEventEnvelopeV2,
+  logger: Logger
+): Promise<void> {
+  await match(decodedMessage)
+    .with(
+      { type: "EServiceDescriptorPublished" },
+      async ({ data: { eservice } }) => {
+        if (eservice) {
+          await handlers.handleNewEServiceVersionPublished(
```

Comment:
here we should understand what to do if the same event could produce different notifications: I think it might be the case if we have to support a different notification if it's the first descriptor for a new eservice (but not sure we'll eventually implement this)

but we can figure that out later if we need to